lib/help_scout/api.rb

```diff
@@ -30,16 +30,18 @@ def put(path, params)
 
     private
 
-    def handle_response(result) # rubocop:disable AbcSize
-      case result.status
-      when 400 then raise BadRequest, result.body&.dig('validationErrors')
-      when 401 then raise NotAuthorized, result.body&.dig('error_description')
-      when 404 then raise NotFound, 'Resource Not Found'
-      when 429 then raise ThrottleLimitReached, result.body&.dig('error')
-      when 500, 501, 503 then raise InternalError, result.body&.dig('error')
+    def handle_response(result) # rubocop:disable AbcSize, Metrics/MethodLength
+      if result.success?
+        HelpScout::Response.new(result)
+      else
+        case result.status
+        when 400 then raise BadRequest, result.body&.dig('validationErrors')
+        when 401 then raise NotAuthorized, result.body&.dig('error_description')
+        when 404 then raise NotFound, 'Resource Not Found'
+        when 429 then raise ThrottleLimitReached, result.body&.dig('error')
+        else raise InternalError, result.body
+        end
       end
-
-      HelpScout::Response.new(result)
     end
 
     def new_connection
```

Comment:
It looks like `HelpScout::API::Client.new` is only called here and from `HelpScout::API::AccessToken.create`. Since we're already instantiating an object here, could be nice to pass the an `authorize` flag to the initializer rather than having two separate methods.

```ruby
attr_reader :authorize
def initialize(authorize: true)
  @authorize = authorize
end

def connection
  @_connection ||= begin
    HelpScout::API::AccessToken.refresh! if authorize && HelpScout.access_token.nil?
    build_connection # no need to pass authorize because it's on the instance
  end
end

private

def build_connection  # we can remove the functional authorize param and instead use reader
  Faraday.new(url: BASE_URL) do |conn|
    conn.request :json
    conn.authorization(:Bearer, HelpScout.access_token.value) if authorize && HelpScout.access_token&.value
    # …
  end
end
```

The nicety is that if we want to conditionally load a connection, as in the example below, there's no need to switch between two method names:

```ruby
def new_connection
  HelpScout::API::Client.new(authorize: authorize).connection
end
```

Reply:
Like it, implemented.